### ISSUE TITLE
EFX and Cluster

### ISSUE THREAD

**Issue by zero-khan** [OldUnreal/UnrealTournamentPatches#67]
I noticed that EFX does not work at all with Cluster. No reverb effect at all. With OpenAL it does work though.

Another difference: OpenAL audio is much much quieter than Cluster. Cluster is way louder.

**Comment by Dr-Flay** [comment]
I have OpenAL selected but can't see the EFX_ZoneInfo in UEd.
Are you testing with an existing map or did you add the actor in a new test ?

**Comment by gargul2** [comment]
Similiar issue: https://github.com/OldUnreal/UnrealTournamentPatches/issues/47

**Comment by zero-khan** [comment]
EFX requires that oldunreal efx.u thing.

**Comment by Dr-Flay** [comment]
If the EFX component is not in the current patch can we assume there is a problem with it ?

**Comment by hanfling** [comment]
EFX package source with the other zoneinfos.
[EFXNerfStuff-20150507.zip](https://github.com/OldUnreal/UnrealTournamentPatches/files/5321523/EFXNerfStuff-20150507.zip)

Also EFX does not requires the EFX.u per se, it just requires the QueryEffects event.

Cluster does not support EFX. So far I haven't thought about implementing it, I might implement it for UT, it's for the most part really just standard I3DL2 listener stuff FMOD also just supports.

**Comment by Dr-Flay** [comment]
I remember that EFX.u was made for compatibility reasons so maps using EFX will work for people without support.
Without it loaded as an editor package there is no EFX actor to place in maps.

@hanfling  Can you clarify if the EFX component in the previous 2.4.7 release of the audio renderer is needed
https://www.oldunreal.com/cgi-bin/yabb2/YaBB.pl?num=1399614690
or if the EFX actor has been moved or is not yet added in the mix

**Comment by hanfling** [comment]
Oh right, we should definetly resuse the exiting EFX.u for UT included in prior ALAudio releases. Didn't came to my mind.

Let me rephrase what I wrote: The EFX.u is an entire optional thing, purely for conveniance. The only thing which does matter is that the QuerryEffects() event inside the ZoneInfo (or it's parent) is present. It was always encouraged (and well basically the idea), that if you use your custom ZoneInfos, that this gets easier to implement. Basically just copy the related code from EFXZoneInfo and adjust to your needs.

But yes, the EFX.u prior shipping with ALAudio should probably be included straight ahead. @stijn-volckaert 

> I remember that EFX.u was made for compatibility reasons so maps using EFX will work for people without support.

The EFXZoneInfo was literally sitting inside the ALAudio package before I rewrote that part, to instead use an (optional) uc event to retrieve the properties from the ZoneInfo. So in case a client also did not have ALAudio installed, it would end up downloading the ALAudio.u from the server, and fail on trying to load it (ALAudio.dll missing) and game would exit.

This also makes it easier for other games which won't want the EFX properties/stuff in the Engine.ZoneInfo. So you can just derive your custom ZoneInfo from WaterZone, etc. If you wanna dig, the original discussion was inside the ALAudio thread you just linked (page 2 or so).

**Comment by garry-ut99** [comment]
> https://github.com/OldUnreal/UnrealTournamentPatches/issues/67#issue-713224171 : zero-khan : Another difference: OpenAL audio is much much quieter than Cluster. Cluster is way louder. 

Yep, a thread with the same issue already exist: https://github.com/OldUnreal/UnrealTournamentPatches/issues/115
